### ISSUE TITLE
Integrate reference "new" reference finder with existing system.

### ISSUE THREAD

**Issue by dantleech** [phpactor/phpactor#1054]
The original reference / class finder evolved in the `class-mover` package and scans the filesystem and uses composer to resolve class FQNs and text filtering to isolate files with class member candidates.

There is now a dedicated API in `phpactor/reference-finders` which is implemented by the `indexer-extension`.

it would be ideal to support both of these strategies and adapt / rewrite the other tooling (e.g. the CLI interface) to use the new reference finder API.

**Comment by elythyr** [comment]
If I recall properly the actual implementation also deals with replacement ?
This is too many responsibilities in my opinion, what about using this opportunity to also change that.

Maybe having a new reference finder that only returns references, with some kind of `Location` that will provide the file and position in it (start/end line and position in the line).
So that it can be used for both "find refenreces" and "replace references", the new replace service will only have to call the "finder" and for each result include the `Location` into some kind of `TextEdit` ?

**Comment by dantleech** [comment]
Yes, there is no reason to couple find references with replacement this time round :)

**Comment by elythyr** [comment]
I was switching for a debian system towards arch and I still have a lot of things to configure before feeling like home in my new setup.
So I'm not sure I would be able to do anything any time soon but I will force myself to come around more often and follows new threads at least.
I have seen some really nice things done in the past month, great job ! I can't wait to try all this out :)

**Comment by dantleech** [comment]
This could be tricky - while we could probably write an implementation based on `ClassMover` the existing functionality won't be easy to refactor/replace.

The new API doesn't support "search" currently - as the target is specified via a file offset.

Maybe this can be deferred until support has been written for search+replace.
